src/Tasks/DownloadFile.cs

```diff
@@ -125,7 +125,8 @@ private async Task<bool> ExecuteAsync()
                     }
                     else
                     {
-                        Log.LogErrorWithCodeFromResources("DownloadFile.ErrorDownloading", SourceUrl, actualException.Message);
+                        Log.LogErrorWithCodeFromResources("DownloadFile.ErrorDownloading", SourceUrl);
+                        Log.LogErrorFromException(actualException, true);
```

Comment:
This makes one error into two errors, right? Can you instead take the output the LogErrorFromException would log and replace actualException.Message with that? (And make sure it looks reasonable, since I haven't checked that.)

Reply:
actualException.Message is original output.  Log.LogErrorWithCodeFromResources("DownloadFile.ErrorDownloading", SourceUrl); is for the sourceUrl info. after that, add the detailed message and stack trace.  Just as image as above.  I use the 'Error message as follows' to make it united. 
![image](https://user-images.githubusercontent.com/26814373/218616601-9c004c56-c008-40f8-a027-39ff59d6a10c.png)

Reply:
I agree that the error looks good; that isn't my concern. My concern is that if you look a little below your screenshot, I suspected it would say 2 errors instead of 1, and indeed it does:
<img width="123" alt="image" src="https://user-images.githubusercontent.com/12969783/218898133-e81acd39-f0ed-439d-8dbe-b5a0a811c078.png">

My suggestion isn't to change the content but to tell MSBuild it's only logging one error so that count is correct.

Reply:
Sorry, I didn't notice that.   Now I change the actualException.Message to actualException.ToString().  The output is 
![image](https://user-images.githubusercontent.com/26814373/218919280-0d98051f-31d4-4c41-99a2-87625f109641.png)